### PR TITLE
Avoid to use to_dataframe in Series.rank to keep anchor

### PR DESCRIPTION
**Pull request by RainFung**
This PR proposes to avoid to use to_dataframe in Series.rank to keep anchor. Currently. `Series.rank` has https://github.com/databricks/koalas/issues/405 issue as well because it converts Spark column into DataFrame within Series API.

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @HyukjinKwon @ueshin

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -2412,7 +2411,43 @@ def rank(self, method='average', ascending=True):
...
+                stat_func = F.min
+            elif method == 'max':
+                stat_func = F.max
+            window1 = Window.orderBy(asc_func(column_name))\
+                .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+            window2 = Window.partitionBy(column_name)\
+                .rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing)
+            scol = stat_func(F.row_number().over(window1)).over(window2)
+        return Series(self._kdf._internal.copy(scol=scol), anchor=self._kdf).rename(column_name)\
+            .astype(np.float64)
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py]: Ditto, can we deduplicate the codes with `DataFrame.rank`? We can call each column, for instnace,

```scala
        applied = []
        for column in self._internal.data_columns:
            applied.append(self[column].rank(...))

        sdf = self._sdf.select(
            self._internal.index_scols + [c._scol for c in applied])
        internal = self._internal.copy(sdf=sdf, data_columns=[c.name for c in applied])
        return DataFrame(internal)
```
```

**Comment by HyukjinKwon** [comment]
It's great that we avoid `to_dataframe` and we keeps the anchor :-).

**Comment by HyukjinKwon** [comment]
@RainFung, let's be explicit about PR title and description next time. I thought you were proposing Series.rank API. I will fix it for this time.

**Pull request merged**